tests/bsim/bluetooth/mesh/CMakeLists.txt

```diff
@@ -48,6 +48,7 @@ elseif(CONFIG_BT_MESH_GATT_PROXY)
 elseif(CONFIG_BT_CTLR_LOW_LAT)
 
  target_sources(app PRIVATE
+  src/friendship_common.c
```

Comment:
It is not required to add this file for any combination if this is common functionality. Friendship is enabled by default in any of our test configurations.

Reply:
Tests are separate apps compiled with overlay, i.e. every config+overlay is it's own app. This must be specified to be built together with any file that wants to use it (some friendship tests run with low latency config, some with base setup, blob and others with mesh1d1 config) That's why friendship.c is listed two times, and this file should be imo treated the same

Reply:
The friend functionality is present in all combinations:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bsim/bluetooth/mesh/prj.conf#L31
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bsim/bluetooth/mesh/prj_mesh1d1.conf#L32
No necessity to play with specific configurations.

Reply:
The way this cmake file seems to be constructed is that if `low_lat` overlay is used, only `  src/test_friendship.c` and 
`src/test_transport.c` are built. `test_friendship.c` requires `src/friendship_common.c` to be built here also. If `src/friendship_common.c` is built for normal and mesh1.1 config, but not for overlay, all functions will be undefined.

Reply:
AFAIK this is why `test_friendship.c` is three times: for mesh, mesh1.1, both with `low_lat` overlay

Reply:
These files are available in all configurations:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bsim/bluetooth/mesh/CMakeLists.txt#L15-L18